lib/Markup/Render/MarkupJekyll.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+
+public struct JekyllGenerator: Render {
+    let permalink: String
+    
+    public init(permalink: String) {
+        self.permalink = permalink
+    }
+
+    public func render(content: String) -> String? {
+        guard let syntaxTree = SyntaxAnalyzer.parse(content: content) else { return nil }
+        let filteredSyntaxTree = syntaxTree.filter { !$0.isHidden }.reduce()
+        //if VERBOSE { syntax.forEach { print($0) } }
+        return filteredSyntaxTree.reduce("") { (acc, node) in acc + node.jekyll(permalink: permalink) }
+    }
+}
+
+// MARK: - Jekyll definition for each node
+extension Node {
+    var isHidden: Bool {
+        switch self {
+        case let .nef(command, _): return command == .hidden
+        default: return false
+        }
+    }
+
+    func jekyll(permalink: String) -> String {
+        switch self {
+        case let .nef(command, nodes):
+            return command.jekyll(nodes: nodes, permalink: permalink)
+
+        case let .markup(_, description):
+            return "\n\(description)"
+
+        case let .block(nodes):
+            let nodesJekyll = nodes.map { $0.jekyll() }.joined()
+            guard !nodesJekyll.isEmpty else { return "" }
+            return "```swift\n\(nodesJekyll)\n```\n"
+            
+        case let .raw(description):
+            return description
+        }
+    }
+}
+
+extension Node.Nef.Command {
+    func jekyll(nodes: [Node], permalink: String) -> String {
+        switch self {
+        case .header:
+            let header = nodes.map{ $0.jekyll(permalink: permalink) }.joined()
+            return """
+            ---
+            \(header)
+            permalink: \(permalink)
+            ---
+            """
+        case .hidden:
+            return ""
+        case .invalid:
+            fatalError("Found .invalid command in nef: \(nodes).")
+        }
+    }
+}
+
+extension Node.Code {
+    func jekyll() -> String {
```

Comment:
`jekyll` seems like something that should be in a protocol, and the extension of `Node.Code` and `Node.Nef.Command` should be conforming to that protocol. I see the signatures are different in the two cases though. Can we think of a way of unifying them?

Reply:
Good point! :) I will review it